docs: create_before_destroy meta-attribute propagation (#32279)
hashicorp/terraform: website/docs/language/meta-arguments/lifecycle.mdx

  such features, so you must understand the constraints for each resource
  type before using `create_before_destroy` with it.

>   Note that Terraform propagates and applies `create_before_destroy` meta-attribute

Reviewer: ```suggestion
  Note that Terraform propagates and applies the `create_before_destroy` meta-attribute
```

>   behaviour to all resource dependencies. For example, if resource A with enabled
>   `create_before_destroy` depends on resource B with disabled `create_before_destroy`
>   (by default), then Terraform enables `create_before_destroy` for resource B
>   implicitly and stores it to the state file. You cannot override `create_before_destroy`
>   to `false` on resource B, because that would imply dependency cycles in the graph.

Reviewer: ```suggestion
  behaviour to all resource dependencies. For example, if `create_before_destroy` is enabled on resource A but not on resource B, but resource A is dependent on resource B, then Terraform enables `create_before_destroy` for resource B 
  implicitly by default and stores it to the state file. You cannot override `create_before_destroy`
  to `false` on resource B because that would imply dependency cycles in the graph.
```
Just a few more tweaks for consistency with our style guidance.

  such features, so you must understand the constraints for each resource
  type before using `create_before_destroy` with it.

  It's important to understand that `create_before_destroy` meta-attribute is

>   stored in state file and _propagated_ to all resource dependencies, including

Reviewer 2: Note for docs team reviewer: "propagated" may not be the correct term to use here.

Author: Whatever reflects the fact of the meta-attribute proliferation through the resource graph 🙂

Reviewer 2: Indeed! That was feedback from triage today, just wanted to capture it for the future.

>   It's important to understand that `create_before_destroy` meta-attribute is
>   stored in state file and _propagated_ to all resource dependencies, including
>   transitive. For example, resource A depends on resource B, hence B can't be
>   destroyed prior to A. The new A depends on the new B, so that has to be made
>   first, hence forcing the B to be `create_before_destroy` as well. You can't
>   override `create_before_destroy` to `false` on B, because the ordering of
>   resource requires it, otherwise there would be dependency cycles in the graph.
> 

Reviewer: ```suggestion
  Note that Terraform stores the `create_before_destroy` meta-attribute in the state file that applies to all resource dependencies. The attribute is transitive, so dependencies for resources with `create_before_destroy` enabled also inherit the `create_before_destroy` behavior. For example, if resource A depends on resource B, then Terraform cannot destroy resource B prior to destroying resource A. If the `create_before_destroy` meta-attribute is enabled for resource A and Terraform creates new instances of resources A and B, then Terraform cannot destroy the previous instance of resource B until it destroys the previous instance of resource A. You cannot
  override `create_before_destroy` to `false` on B because the order of
  resource creation and destruction requires it. Otherwise, there would be dependency cycles in the graph.

```
I tried to make the explanation more explicit. As is, we're implying the behavior through an example. Please let me know if I misunderstood and changed the meaning.

Author: Well, I'm not native English speaking person, but **propagation** feels like a more correct term reckoning my university course 😅 

**Attribute propagation** involves the transfer or propagation (verb) of attributes from one entity to another within a system. It is a process where an attribute of an entity is **passed or copied** to another object or entity. The goal of attribute propagation is to ensure that relevant information or characteristics are transmitted and maintained across different components or entities. That's exactly what I observed Terraform has done – copied `create_before_destroy` to all objects in the chain.

**Attribute transitive** refers to the transitive property (noun) of attributes in a system. Transitivity means that if attribute A is related to attribute B, and attribute B is related to attribute C, then attribute A can be considered indirectly related to attribute C. In other words, the transitive property allows attributes to be inherited or derived through a chain of relationships.

Anyway, I'm OK with your variant. The most important – pay attention of a regular user to this intrinsic behaviour.

Reviewer: Thanks for the explanation, @Tensho 
Do you mind committing the suggestion if it makes sense so that we can merge this PR? We appreciate your help!

Author: Done
